cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/DatasetServiceStore.java

```diff
@@ -64,4 +72,41 @@ protected void startUp() throws Exception {
   protected void shutDown() throws Exception {
     table.close();
   }
+
+  @Override
+  public synchronized void setRestartInstancesRequest(String serviceName, long startTimeMs, long endTimeMs,
+                                                      boolean isSuccess, Integer instanceId) {
+    RestartServiceInstancesStatus.RestartStatus status = RestartServiceInstancesStatus.RestartStatus.SUCCESS;
```

Comment:
Why not use `status = isSuccess ? SUCCESS : FAILURE` instead of setting then setting again conditionally?